plugin-server/src/worker/ingestion/event-pipeline/hog-transformations/hog-transformer.service.ts

```diff
@@ -0,0 +1,174 @@
+    }
+
+    public async start(): Promise<void> {
+        const hogTypes: HogFunctionTypeType[] = ['transformation']
+        await this.hogFunctionManager.start(hogTypes)
+    }
+
+    public transformEvent(event: PluginEvent): Promise<PluginEvent> {
+        return runInstrumentedFunction({
+            statsKey: `hogTransformer`,
+            // there is no await as all operations are sync
+            // eslint-disable-next-line @typescript-eslint/require-await
+            func: async () => {
+                const teamHogFunctions = this.hogFunctionManager.getTeamHogFunctions(event.team_id)
+                const transformationFunctions = this.getTransformationFunctions()
+                // For now, execute each transformation function in sequence
+                // Later we can add support for chaining/ordering
+                for (const hogFunction of teamHogFunctions) {
+                    const invocation = this.createHogFunctionInvocation(event, hogFunction)
+                    const result = this.hogExecutor.execute(invocation, { functions: transformationFunctions })
+                    if (result.error) {
+                        status.warn('⚠️', 'Error in transformation', {
+                            error: result.error,
+                            function_id: hogFunction.id,
+                            team_id: event.team_id,
+                        })
+                        continue
+                    }
+
+                    // Type check execResult before accessing result
+                    if (!result.execResult) {
+                        status.warn('⚠️', 'Missing execution result - no transformation applied')
+                        return event
+                    }
+
+                    const transformedEvent: unknown = result.execResult
+
+                    // Validate the transformed event has a properties object
+                    if (
+                        !transformedEvent ||
+                        typeof transformedEvent !== 'object' ||
+                        !('properties' in transformedEvent)
+                    ) {
+                        status.warn('⚠️', 'Invalid transformation result - missing properties', {
+                            function_id: hogFunction.id,
+                        })
+                        continue
+                    }
+
+                    // Validate properties are of correct type
+                    if (!this.validateProperties(transformedEvent.properties)) {
+                        status.warn('⚠️', 'Invalid transformation result - invalid properties', {
+                            function_id: hogFunction.id,
+                            properties: transformedEvent.properties,
+                        })
+                        return event
+                    }
+
+                    // Validate event name is a string if present
+                    if ('event' in transformedEvent && typeof transformedEvent.event !== 'string') {
```

Comment:
Unless I'm misreading the event isn't supported? You should be allowed to modify the event name